plugins/org.eclipse.n4js.ide/src/org/eclipse/n4js/ide/xtext/server/XWorkspaceManager.java

```diff
@@ -97,6 +99,38 @@
 
 	private final Map<URI, XDocument> openDocuments = new ConcurrentHashMap<>();
 
+	/** See {@link #addFilesAwaitingGeneration(Collection, Collection)}. */
+	private final FileChangeTracker filesAwaitingGeneration = new FileChangeTracker();
+
+	/** Thread-safe tracking of file changes and/or deletions over time. */
+	private static class FileChangeTracker {
+
+		private final Set<URI> changedFiles = new LinkedHashSet<>();
+		private final Set<URI> deletedFiles = new LinkedHashSet<>();
+
+		public synchronized void clear() {
```

Comment:
The synchronized keywords suggest that this code is called from different threads. Is that the case?

Reply:
It was, but it doesn't matter anymore because due to moving this to XBuildManager it is no longer necessary.